File: app/src/main/java/com/ytempest/wanandroid/activity/main/knowledge/KnowledgePresenter.java
package com.ytempest.wanandroid.activity.main.knowledge;

import androidx.annotation.NonNull;

import com.ytempest.wanandroid.base.presenter.BasePresenter;
import com.ytempest.wanandroid.http.bean.KnowledgeArchitectureBean;
import com.ytempest.wanandroid.http.observer.HandlerObserver;
import com.ytempest.wanandroid.interactor.impl.BaseInteractor;
import com.ytempest.wanandroid.utils.RxUtils;

import java.util.List;

import javax.inject.Inject;

/**
 * @author heqidu
 * @since 2020/6/30
 */
public class KnowledgePresenter extends BasePresenter<IKnowledgeContract.View> implements IKnowledgeContract.Presenter {

    @Inject
    public KnowledgePresenter(BaseInteractor interactor) {
        super(interactor);
    }

    @Override
    public void getKnowledgeArchitecture() {
        mInteractor.getHttpHelper().getKnowledgeArchitecture()
                .compose(RxUtils.switchScheduler())
                .subscribe(new HandlerObserver<List<KnowledgeArchitectureBean>>(mView) {
                    @Override
                    protected void onSuccess(@NonNull List<KnowledgeArchitectureBean> list) {
                        mView.onKnowledgeArchitectureReceived(list);
                    }

                    @Override
                    protected void onFail(int code, Throwable e) {
                        super.onFail(code, e);
                        mView.onKnowledgeArchitectureFail(code);
                    }
                });
    }
}

File: app/src/main/java/com/ytempest/wanandroid/activity/main/knowledge/KnowledgePresenter.kt
package com.ytempest.wanandroid.activity.main.knowledge

import com.ytempest.wanandroid.base.presenter.BasePresenter
import com.ytempest.wanandroid.http.observer.HandlerObserver
import com.ytempest.wanandroid.interactor.impl.BaseInteractor
import com.ytempest.wanandroid.utils.RxUtils
import com.ytempest.wanandroid.http.bean.KnowledgeArchitectureBean
import javax.inject.Inject

/**
 * @author heqidu
 * @since 21-2-9
 */
class KnowledgePresenter @Inject constructor(
        interactor: BaseInteractor
) : BasePresenter<IKnowledgeView>(interactor), IKnowledgePresenter {

    override fun getKnowledgeArchitecture() {
        mInteractor.getHttpHelper().getKnowledgeArchitecture()
                .compose(RxUtils.switchScheduler())
                .subscribe(object : HandlerObserver<List<KnowledgeArchitectureBean>>(mView) {
                    override fun onSuccess(list: List<KnowledgeArchitectureBean>) {
                        mView.onKnowledgeArchitectureReceived(list)
                    }

                    override fun onFail(code: Int, e: Throwable) {
                        super.onFail(code, e)
                        mView.onKnowledgeArchitectureFail(code)
                    }
                })
    }

}